### PR TITLE
[AWS] Add regions requested by the user

### PR DESCRIPTION
**Pull request by Michaelvll**
Our user requested to use `ap-east-1` region, we now enable this for the user so that the auto-updated catalog works.

### CODE REVIEW

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 1]
```diff
@@ -31,10 +31,10 @@
     # 'me-south-1',
     # 'me-central-1',
     # 'af-south-1',
-    # 'af-south-2',
```
Comment (WoosukKwon) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:34]: nit: why is this deleted?

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 2]
```diff
@@ -31,10 +31,10 @@
...
-    # 'ap-east-1',
+    'ap-east-1',  # enable this non-default region due to user request
     # 'ap-southeast-3',
     'ap-south-1',
+    # 'ap-south-2',
```
Comment (WoosukKwon) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:37]: Is this a new region? or why is it added now?

**Review by WoosukKwon — COMMENTED** [review]
@Michaelvll Thanks for this PR. Could you add how we selected the supported non-US regions so far and what we should do to enable new regions?

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:34]: There is no` af-south-2`. This should be a typo for `ap-south-2`?

**Comment by Michaelvll** [comment]
> Could you add how we selected the supported non-US regions so far and what we should do to enable new regions?

At the beginning of the least we had a comment for how we select the regions.
https://github.com/skypilot-org/skypilot/blob/f31859ec4c4d9c588c16fc0116b5e08e65a80604/sky/clouds/service_catalog/data_fetchers/fetch_aws.py#L16

Do you think it is not clear enough?

File: sky/clouds/service_catalog/aws_catalog.py [hunk 3]
```diff
@@ -52,7 +52,10 @@ def _apply_az_mapping(df: 'pd.DataFrame') -> 'pd.DataFrame':
         az_mappings.to_csv(az_mapping_path, index=False)
     else:
         az_mappings = pd.read_csv(az_mapping_path)
-    df = df.merge(az_mappings, on=['AvailabilityZone'], how='left')
+    # Use inner join to drop rows with unknown AZ IDs, which are likely
+    # because the user does not have access to that Region. Otherwise,
+    # there will be rows with NaN in the AvailabilityZone column.
+    df = df.merge(az_mappings, on=['AvailabilityZone'], how='inner')
```
Comment (Michaelvll) [inline sky/clouds/service_catalog/aws_catalog.py:58]: This is a necessary fix, for the user who do not have `ap-east-1` enabled in her AWS account. Otherwise, the AZ column will be NaN, causing failure in our failover codes.

Comment (concretevitamin) [reply sky/clouds/service_catalog/aws_catalog.py:58]: This may also fix #1459? cc @iojw to test it out once this PR is merged.

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (concretevitamin) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:34]: Can we fix L33 as well?

**Review by concretevitamin — APPROVED** [review]
LGTM

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:34]: I think L33 is correct. AWS does have `af-south-1`, but no `af-south-2`.

File: sky/clouds/service_catalog/aws_catalog.py (hunk 3 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/aws_catalog.py:58]: Good point! This should fix the problem.

Comment (iojw) [reply sky/clouds/service_catalog/aws_catalog.py:58]: I'm still seeing the same error for the `test-image-id-dict-with-zone` smoke test on the latest branch. Is there anything I need to do after merging?

Comment (Michaelvll) [reply sky/clouds/service_catalog/aws_catalog.py:58]: Could you try to `rm -r ~/.sky/catalogs/v5/aws` and try that test again?

Comment (iojw) [reply sky/clouds/service_catalog/aws_catalog.py:58]: The same error occurs

```
❯ rm -r ~/.sky/catalogs/v5/aws
❯ bash tests/run_smoke_tests.sh test_image_id_dict_with_zone
I 12-08 21:10:40 aws_catalog.py:47] Fetching availability zones mapping for AWS...
bringing up nodes...
[image_id_dict_with_region] Test started. Log: less /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/T/image_id_dict_with_region-8rxmgc4j.log
[image_id_dict_with_region] Failed.
[image_id_dict_with_region] Reason: sky launch -y -c test-image-id-dict-with-zone-a64f6128-5a --zone us-west-1a examples/per_region_images.yaml
[image_id_dict_with_region] Log: less /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/T/image_id_dict_with_region-8rxmgc4j.log
[image_id_dict_with_region]
F
================================================================ FAILURES ================================================================
______________________________________________________ test_image_id_dict_with_zone ______________________________________________________
[gw1] darwin -- Python 3.9.15 /Users/isaac/.local/share/virtualenvs/sky-b0s4-8BH/bin/python
tests/test_smoke.py:262: in test_image_id_dict_with_zone
    run_one_test(test)
tests/test_smoke.py:117: in run_one_test
    raise Exception(f'test failed: less {log_file.name}')
E   Exception: test failed: less /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/T/image_id_dict_with_region-8rxmgc4j.log
======================================================== short test summary info =========================================================
FAILED tests/test_smoke.py::test_image_id_dict_with_zone - Exception: test failed: less /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/...
1 failed, 368 warnings in 16.79s
```

```
Last login: Thu Dec  8 21:11:07 on ttys008
❯ cat /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/T/image_id_dict_with_region-77ulvuux.log
+ sky launch -y -c test-image-id-dict-with-zone-a64f6128-b0 --zone us-west-2b examples/per_region_images.yaml && exit 1 || true
Task from YAML spec: examples/per_region_images.yaml
W 12-08 20:51:39 resources.py:654] image_id in resources is experimental. It only supports AWS/GCP.
ValueError: image_id {'us-east-1': 'ami-0729d913a335efca7', 'us-west-1': 'skypilot:gpu-ubuntu-1804'} should contain the image for the specified region us-west-2.
+ sky status | grep test-image-id-dict-with-zone-a64f6128-b0 && exit 1 || true
+ sky launch -y -c test-image-id-dict-with-zone-a64f6128-b0 --zone us-west-1a examples/per_region_images.yaml
Task from YAML spec: examples/per_region_images.yaml
W 12-08 20:51:40 resources.py:654] image_id in resources is experimental. It only supports AWS/GCP.
D 12-08 20:51:41 optimizer.py:232] #### Task(run='conda env list\n')
D 12-08 20:51:41 optimizer.py:232]   resources: {AWS(None, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'})} ####
D 12-08 20:51:41 optimizer.py:261] Defaulting the task's estimated time to 1 hour.
D 12-08 20:51:41 optimizer.py:278] resources: AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'})
D 12-08 20:51:41 optimizer.py:287]   estimated_runtime: 3600 s (1.0 hr)
D 12-08 20:51:41 optimizer.py:291]   estimated_cost (not incl. egress): $0.4
I 12-08 20:51:41 optimizer.py:606] == Optimizer ==
I 12-08 20:51:41 optimizer.py:629] Estimated cost: $0.4 / hour
I 12-08 20:51:41 optimizer.py:629]
I 12-08 20:51:41 optimizer.py:685] Considered resources (1 node):
I 12-08 20:51:41 optimizer.py:714] ------------------------------------------------------------------
I 12-08 20:51:41 optimizer.py:714]  CLOUD   INSTANCE      vCPUs   ACCELERATORS   COST ($)   CHOSEN
I 12-08 20:51:41 optimizer.py:714] ------------------------------------------------------------------
I 12-08 20:51:41 optimizer.py:714]  AWS     m6i.2xlarge   8       -              0.38          ✔
I 12-08 20:51:41 optimizer.py:714] ------------------------------------------------------------------
I 12-08 20:51:41 optimizer.py:714]
Running task on cluster test-image-id-dict-with-zone-a64f6128-b0...
I 12-08 20:51:41 cloud_vm_ray_backend.py:2920] Creating a new cluster: "test-image-id-dict-with-zone-a64f6128-b0" [1x AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'})].
I 12-08 20:51:41 cloud_vm_ray_backend.py:2920] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 12-08 20:51:41 cloud_vm_ray_backend.py:1001] To view detailed progress: tail -n100 -f /Users/isaac/sky_logs/sky-2022-12-08-20-51-40-980965/provision.log
I 12-08 20:51:41 cloud_vm_ray_backend.py:1257] Launching on AWS us-west-1 (us-west-1a)
D 12-08 20:51:47 cloud_vm_ray_backend.py:1336] Ray up takes 6.213129043579102 seconds with 1 retries.
I 12-08 20:51:47 cloud_vm_ray_backend.py:686] ====== stdout ======
2022-12-08 20:51:41,927	INFO commands.py:289 -- Cluster: test-image-id-dict-with-zone-a64f6128-b0
2022-12-08 20:51:42,157	INFO commands.py:366 -- Checking External environment settings
2022-12-08 20:51:42,157	VINFO utils.py:149 -- Creating AWS resource `ec2` in `us-west-1`
2022-12-08 20:51:46,396	VINFO utils.py:149 -- Creating AWS resource `iam` in `us-west-1`
2022-12-08 20:51:46,881	VINFO utils.py:149 -- Creating AWS resource `ec2` in `us-west-1`

I 12-08 20:51:47 cloud_vm_ray_backend.py:689] ====== stderr ======
2022-12-08 20:51:41,927	WARNING util.py:241 -- Dropping the empty legacy field head_node. head_nodeis not supported for ray>=2.0.0. It is recommended to removehead_node from the cluster config.
2022-12-08 20:51:41,927	WARNING util.py:241 -- Dropping the empty legacy field worker_nodes. worker_nodesis not supported for ray>=2.0.0. It is recommended to removeworker_nodes from the cluster config.
2022-12-08 20:51:41,927	INFO util.py:364 -- setting max workers for head node type to 0
2022-12-08 20:51:47,199	PANIC config.py:517 -- No usable subnets matching availability zone ['us-west-1a'] found for node type ray.head.default.
Choose a different availability zone or try manually creating an instance in your specified region to populate the list of subnets and trying this again.
Error: No usable subnets matching availability zone ['us-west-1a'] found for node type ray.head.default.
Choose a different availability zone or try manually creating an instance in your specified region to populate the list of subnets and trying this again.

Clusters
NAME                                      LAUNCHED        RESOURCES                                                                  STATUS  AUTOSTOP  COMMAND
test-image-id-dict-with-zone-a64f6128-b0  a few secs ago  1x AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'}...  INIT    -         sky launch -y -c test-ima...
test-image-id-dict-with-zone-a64f6128-b5  1 hr ago        1x AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'}...  INIT    -         sky launch -y -c test-ima...
test-image-id-dict-with-zone-a64f6128-9b  2 hrs ago       1x AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'}...  INIT    -         sky launch -y -c test-ima...
test-image-id-dict-with-zone-a64f6128-bd  2 hrs ago       1x AWS(m6i.2xlarge, image_id={'us-west-1': 'skypilot:gpu-ubuntu-1804'}...  INIT    -         sky launch -y -c test-ima...

Managed spot controller (autostopped if idle for 10min)
Use spot jobs CLI: sky spot --help
NAME                          LAUNCHED  RESOURCES                          STATUS   AUTOSTOP  COMMAND
sky-spot-controller-a64f6128  1 hr ago  1x AWS(m6i.2xlarge, disk_size=50)  STOPPED  10m       sky spot launch -n test-s...

RuntimeError: Errors occurred during provision; check logs above.
Failed.
Reason: sky launch -y -c test-image-id-dict-with-zone-a64f6128-b0 --zone us-west-1a examples/per_region_images.yaml
Log: less /var/folders/m_/gym1_4y53m95hb50x8qc1myw0000gn/T/image_id_dict_with_region-77ulvuux.log
```
